javascript/code-challenges/challenges-03.test.js

```diff
@@ -0,0 +1,303 @@
+
+For example, oddValues([1,2,3]) returns [1,3].
+------------------------------------------------------------------------------------------------ */
+
+const oddValues = (arr) => {
+  return arr.filter(num => num % 2);
+};
+
+/* ------------------------------------------------------------------------------------------------
+CHALLENGE 5
+
+Write a function named notInFirstArray that, given two arrays as input, uses filter to return an array of all the elements in the second array that are not included in the first array.
+
+For example, notInFirstArray([1,2,3], [1,2,3,4]) returns [4].
+------------------------------------------------------------------------------------------------ */
+
+const notInFirstArray = (forbiddenValues, arr) => {
+  return arr.filter(value => !(forbiddenValues.includes(value)));
+};
+
+/* ------------------------------------------------------------------------------------------------
+CHALLENGE 6 - Stretch Goal
+
+Write a function named getBaseStatGreaterThan that, given the snorlaxData, below, and an integer as input, uses filter to return an array containing all stats with a baseStat greater than the integer.
+
+For example, getBaseStatGreaterThan(snorlaxData.stats, 50) will return an array containing the 'special-defense' and 'special-attack' objects.
+------------------------------------------------------------------------------------------------ */
+
+const snorlaxData = {
+  stats: [
+    {
+      stat: {
+        url: 'https://pokeapi.co/api/v2/stat/6/',
+        name: 'speed',
+      },
+      effort: 5,
+      baseStat: 30,
+    },
+    {
+      stat: {
+        url: 'https://pokeapi.co/api/v2/stat/5/',
+        name: 'special-defense',
+      },
+      effort: 2,
+      baseStat: 110,
+    },
+    {
+      stat: {
+        url: 'https://pokeapi.co/api/v2/stat/4/',
+        name: 'special-attack',
+      },
+      effort: 9,
+      baseStat: 65,
+    },
+  ],
+  name: 'snorlax',
+  weight: 4600,
+};
+
+const getBaseStatGreaterThan = (arr, minBaseStat) => {
+  return arr.filter(abilityData => abilityData.baseStat > minBaseStat);
+};
+
+/* ------------------------------------------------------------------------------------------------
+CHALLENGE 7 - Stretch Goal
+
+Write a function named getStatName that is an extension of your getBaseStatGreaterThan function from challenge 7. For this function, extend your solution from challenge 7 to only return the name of the stat, rather than the entire stat object.
+
+For example, getStatName(snorlaxData.stats, 50) will return ['special-defense', 'special-attack'].
+------------------------------------------------------------------------------------------------ */
+
+const getStatName = (arr, minBaseStat) => {
+  const viableData = arr.filter(abilityData => abilityData.baseStat > minBaseStat);
+  return viableData.map(viable => viable.stat.name);
```

Comment:
So a cool thing you could have done here was chain your functions. Since you already did the work of "viableData" in your previous function, no need to do it again. Instead do something like this
```js
const getStatName = (arr, minBaseStat) => {
  return getBaseStatGreaterThan(arr.minBaseStat).map(viableStat => viableStat.stat.name);
}
```